lib/presentation/model/file/file_asset_entity.dart

```diff
@@ -0,0 +1,47 @@
+import 'dart:typed_data';
+
+import 'package:equatable/equatable.dart';
+import 'package:fluffychat/presentation/model/file/image_asset_entity.dart';
+import 'package:fluffychat/presentation/model/file/video_asset_entity.dart';
+import 'package:matrix/matrix.dart';
+import 'package:photo_manager/photo_manager.dart';
+
+class FileAssetEntity with EquatableMixin {
+
+  final AssetEntity assetEntity;
+
+  FileAssetEntity({required this.assetEntity});
+
+  Future<FileInfo?> toFileInfo() {
+    throw UnimplementedError('FileAssetEntity have not been implemented.');
+  }
+
+  Future<MatrixFile?> toMatrixFile() {
+    throw UnimplementedError('FileAssetEntity have not been implemented.');
+  }
```

Comment:
why we need it @@?

Reply:
We need these methods, because, image and video, and later maybe audio, have different implementations when convert to `MatrixFile` and `FileInfo`. `Throw` will be catch by the interactor.